tests/Doctrine/Tests/DBAL/Functional/ExternalPDOInstanceTest.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Doctrine\Tests\DBAL\Functional;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Driver\PDO\SQLite\Driver as PDOSqliteDriver;
+use Doctrine\DBAL\FetchMode;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\Tests\DbalTestCase;
+use PDO;
+
+class ExternalPDOInstanceTest extends DbalTestCase
+{
+    /** @var Connection */
+    protected $connection;
+
+    protected function setUp(): void
+    {
+        $pdo = new PDO('sqlite::memory:');
```

Comment:
> Actually, I've had to remove this check as I'm not extending DbalFunctionalTestCase anymore, so I don't have access to the currently configured driver.

What will happen here if a run the test suite with say `sqlsrv` configuration and don't have `pdo_sqlite` installed? All existing functional tests have proper checks for the installed extensions and current configuration. Why do you need it to not be a functional one?

> May I'd need to re-introduce a condition before running these tests, either reading from the global config, or at least check that PDO SQLite is installed?

I'd write a proper functional test that:
1. Skips if the currently used connection is not PDO.
2. Extract PDO from the connection.
3. Create a new connection with external PDO.
4. Proceed with testing.

Reply:
Good point, I forgot that `PDOConnection` *extends* `PDO` in 2.x, so I can indeed get the PDO instance from the existing `Connection` to create a new one. Done!

Reply:
Actually, while the tests do pass, **they also pass on 2.10** now I'm afraid. This is because the PDO we extract from the Connection is a PDOConnection, not a raw PDO. So we're back to square one: there does not seem to be a way to run this test for all PDO drivers, without creating our own PDO instances, which requires duplicating the PDO creation logic from the driver (DSN etc.) in the tests.

This is what I called *overkill* in development time, for a quick fix that will be gone in 3.0 anyway.

Reply:
> This is because the PDO we extract from the Connection is a PDOConnection, not a raw PDO.

Uh... you're right. This is why we're getting rid of this untestable inheritance. Please restore the previous version and squash.

> This is what I called _overkill_ in development time

I can't agree. Fixing something that is not supposed to be used may be overkill. Proper testing is not.

Reply:
> Please restore the previous version and squash.

Done! And re-introduced the PDO SQLite driver check to only run tests on CI that target SQLite, and prevent it from running when SQLite might not be available.